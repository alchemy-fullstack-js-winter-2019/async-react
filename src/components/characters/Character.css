.Character {
    border: 3px solid orangered;
}