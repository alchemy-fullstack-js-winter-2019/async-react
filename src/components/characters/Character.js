import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class Character extends PureComponent {
  static propTypes = {
    result: PropTypes.object.isRequired
  }

  render() {
    const { image, name, status, species, gender } = this.props.result;
    
    return (
      <li>
        <img src={image}/>
        <p>{name}</p>
        <p>Status: {status}</p>
        <p>Species: {species}</p>
        <p>Gender: {gender}</p>
      </li>
    );
  }
}
