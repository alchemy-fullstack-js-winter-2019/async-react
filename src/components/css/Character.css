.char {
  background-color: grey;
  text-align: center;
  margin: 5px;
}



