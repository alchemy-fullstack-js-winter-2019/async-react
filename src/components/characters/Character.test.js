import React from 'react';
import renderer from 'react-test-renderer';
import Character from './Character';

describe('Character', () => {
  it('matches a snapshot', () => {
    const character = {
      name: 'bob',
      img: 'https://www.littledayout.com/wp-content/uploads/articles/2017/07-July/20170715-WorldEmojiDay/DabEmoji.jpg',
      status: 'alive',
      species: 'human',
      gender: 'mail'
    };

    const tree = renderer.create(
      <Character
        result={character}
      />
    ).toJSON();
  
    expect(tree).toMatchSnapshot();
  });
});
