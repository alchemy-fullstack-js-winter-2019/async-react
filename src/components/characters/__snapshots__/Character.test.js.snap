// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character matches a snapshot 1`] = `
<li>
  <img />
  <p>
    bob
  </p>
  <p>
    Status: 
    alive
  </p>
  <p>
    Species: 
    human
  </p>
  <p>
    Gender: 
    mail
  </p>
</li>
`;
